app/main.py

```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}   # class attribute
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name  # instance attribute
+        self.age = age  # instance attribute
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = [Person(person["name"], person["age"]) for person in people]
+    for person in people:
+        new_key = "wife" if "wife" in person else "husband"
```

Comment:
The logic for determining `new_key` assumes that a person can only have either a 'wife' or a 'husband', but not both. Ensure this assumption aligns with your data model requirements.